action/sealedenvelope.go

```diff
@@ -179,3 +183,27 @@ func wrapStakingActionIntoExecution(ab AbstractAction, toAddr []byte, pb proto.M
 		data:           data,
 	}, nil
 }
+
+// Verify verifies the action using sender's public key
+func (sealed *SealedEnvelope) Verify() error {
```

Comment:
it is hard to say whether it is appropriate to have this function as a function of struct sealedenvelope for the following reasons:
1. It uses so many public functions of itself, which is not recommended in our codebase
2. The original `Verify` function is more like a util function, which groups some checks together. It is only used in `action/protocol/generic_validator.go`. The usage of this function in several test files are misuse. We can split it into several checks:
a. check public key
b. check gas limit
c. check signature
check a and b could be moved into `action/protocol/generic_validator.go`